src/autopas/tuning/tuningStrategy/BayesianSearch.h

```diff
@@ -26,7 +27,7 @@ namespace autopas {
  * to a Gaussian Process. This allows to estimate the 'gain' of testing a given
  * feature next.
  */
-class BayesianSearch : public TuningStrategyInterface {
+class BayesianSearch final : public TuningStrategyInterface {
```

Comment:
I see it like this:
Currently, we don't intend to inherit from this class. Best case, the compiler can eliminate some virtual calls. This is inside the library, so no "user" should tamper with this anyway. If any developer feels the need to inherit from this class, they can remove the final if they like.

So no, I haven't measured the impact, but I also have no strong opinion about it being there. Just that it currently gives us more potential benefits than not being there.

Reply:
Okay, I agree in this case, we can leave it there since it is internal of the library.
I first encountered the `final` keywords in our functors, which are exposed to the outside user and I think we should remove it there. Not part of this PR of course, but hope you'd agree :D